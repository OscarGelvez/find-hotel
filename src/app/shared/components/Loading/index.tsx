import * as React from 'react';
import Spinner from '../../../../assets/img/loading.gif';

/**
 * Componente encargado de renderizar el spinner de carga
 */
const Loading = () => {
  return (
    <>
      <div data-testid="view-spinner" className="container-loader">
        <img
          data-testid="view-spinner-loader"
          src={Spinner}
          alt="loading"
          className="img-loader"
        />
      </div>
    </>
  );
};
export default Loading;
