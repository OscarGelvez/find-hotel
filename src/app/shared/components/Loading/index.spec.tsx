import '@testing-library/jest-dom';

import { cleanup, render } from '@testing-library/react';
import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

import Loading from './index';

afterEach(cleanup);

it('rederizar el componente Loading sin errores', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Loading />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renderiza el componente Loading y verifica existencia de componentes hijos', () => {
  const { getByTestId } = render(<Loading />);

  expect(getByTestId('view-spinner')).toContainElement(
    getByTestId('view-spinner-loader')
  );
});

it('Compara snapshot del componente Loading en modo lista', () => {
  const element = renderer
    .create(
      <BrowserRouter>
        <Loading />
      </BrowserRouter>
    )
    .toJSON();
  expect(element).toMatchSnapshot();
});
