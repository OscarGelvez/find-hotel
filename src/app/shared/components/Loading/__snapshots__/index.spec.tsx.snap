// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compara snapshot del componente Loading en modo lista 1`] = `
<div
  className="container-loader"
  data-testid="view-spinner"
>
  <img
    alt="loading"
    className="img-loader"
    data-testid="view-spinner-loader"
    src="loading.gif"
  />
</div>
`;
